db/migrations/00034_api_extensions.sql

```diff
@@ -0,0 +1,56 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TYPE extension_status AS ENUM('online', 'offline');
+CREATE TYPE extension_resource_scope AS ENUM('user', 'system');
+
+CREATE TABLE extensions (
+  id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+  name STRING NOT NULL,
+  description STRING NOT NULL,
+  enabled BOOL NOT NULL,
+  slug STRING NOT NULL,
+	status extension_status NOT NULL DEFAULT 'offline',
+
+  created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  deleted_at TIMESTAMPTZ NULL,
+
+  CONSTRAINT extension_slug_key UNIQUE (slug) WHERE deleted_at IS NULL,
+  INDEX (slug, deleted_at) STORING (name, description, enabled, status, created_at, updated_at),
+  INDEX (deleted_at) STORING (name, description, enabled, slug, status, created_at, updated_at)
+);
+
+CREATE TABLE extension_resource_definitions (
+  id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+  name STRING NOT NULL,
+  description STRING NOT NULL,
+  enabled BOOL NOT NULL DEFAULT true,
+
+  slug_singular STRING NOT NULL,
+  slug_plural STRING NOT NULL,
```

Comment:
do we need to store two slugs?

Reply:
we will use the singular to identify the ERD itself, and use the plural for the URIs of the extension resources

Reply:
but cant we just generate one from the other? or is it hard bc its been slug-ified already?

Reply:
yeah but we also have `goose` vs `geese` and `moose` vs `moose`, I'd say the extension developer should take care of this

Reply:
thats fine with me - there are actually libraries to do that i think?